src/main/java/dev/dkvl/womutils/WomUtilsPlugin.java

```diff
@@ -205,6 +249,259 @@ public void onResponse(Call call, Response response)
+		Type messageType = new TypeToken<JsonObject>() {}.getType();
+		String responseBody = response.body().string();
+		JsonObject data = gson.fromJson(responseBody, messageType);
+		String errorMessage = data.get("message").toString().replace("\"", "");
+
+		if (!add && errorMessage.contains("Successfully"))
+		{
+			return;
+		}
+
+		ChatMessageBuilder cmb = new ChatMessageBuilder();
+		cmb.append(ChatColorType.HIGHLIGHT).append(errorMessage);
+
+		chatMessageManager.queue(QueuedMessage.builder()
+				.type(ChatMessageType.CONSOLE)
+				.runeLiteFormattedMessage(cmb.build())
+				.build());
+	}
+
+	private void sendMembersRequest(Request request)
+	{
+		okHttpClient.newCall(request).enqueue(new Callback()
+		{
+			@Override
+			public void onFailure(Call call, IOException e)
+			{
+				log.warn("Error while fetching members list {}.", e.getMessage());
+			}
+
+			@Override
+			public void onResponse(Call call, Response response) throws IOException {
+				groupMembers.clear();
+				Type typeOfArrayList = new TypeToken<ArrayList<GroupMember>>() {}.getType();
+				String responseBody = response.body().string();
+				ArrayList<GroupMember> data = gson.fromJson(responseBody, typeOfArrayList);
+				response.close();
+
+				for (GroupMember gm : data) {
+					groupMembers.add(gm.getUsername());
+				}
+			}
+		});
+	}
+
+	@Subscribe
+	public void onMenuEntryAdded(MenuEntryAdded event)
+	{
+		if (!config.menuOptions())
+		{
+			return;
+		}
+
+		int groupId = WidgetInfo.TO_GROUP(event.getActionParam1());
+		String option = event.getOption();
+		String target = event.getTarget();
+
+		if (groupId == WidgetInfo.FRIENDS_CHAT.getGroupId()
+				|| groupId == WidgetInfo.FRIENDS_LIST.getGroupId())
+		{
+
+			if (!AFTER_OPTIONS.contains(option))
+			{
+				return;
+			}
+
+			final MenuEntry addMember = new MenuEntry();
+			if (groupMembers.contains(Text.toJagexName(Text.removeTags(target).toLowerCase())))
+			{
+				addMember.setOption(REMOVE_MEMBER);
+			} else {
+				addMember.setOption(ADD_MEMBER);
+			}
+			addMember.setTarget(target);
+			addMember.setType(MenuAction.RUNELITE.getId());
+			addMember.setParam0(event.getActionParam0());
+			addMember.setParam1(event.getActionParam1());
+			addMember.setIdentifier(event.getIdentifier());
+
+			insertMenuEntry(addMember, client.getMenuEntries());
+		}
+	}
+
+	@Subscribe
+	public void onPlayerMenuOptionClicked(PlayerMenuOptionClicked event)
+	{
+		String target = Text.toJagexName(event.getMenuTarget()).toLowerCase();
+		if (event.getMenuOption().equals(ADD_MEMBER))
+		{
+			addGroupMember(target);
+		}
+		else if (event.getMenuOption().equals(REMOVE_MEMBER))
+		{
+			removeGroupMember(target);
+		}
+	}
+
+	@Subscribe
+	public void onWidgetMenuOptionClicked(final WidgetMenuOptionClicked event)
+	{
+		WidgetInfo widget = event.getWidget();
+		if (widget == WidgetInfo.FIXED_VIEWPORT_FRIENDS_CHAT_TAB
+				|| widget == WidgetInfo.RESIZABLE_VIEWPORT_FRIENDS_CHAT_TAB)
```

Comment:
This will do a request everytime someone clicks on their friends tab, is that a lot? It should be ok right..?

Reply:
Turn out this doesn't fire for non-menumanager options, but it should still check the entry, in case of some other plugin decides to use the widget for something.